apps/scan/backend/src/store.ts

```diff
@@ -941,16 +942,18 @@ export class Store {
   /**
    * Stores the name of the directory that we'll be continuously exporting to
    */
-  setExportDirectoryName(exportDirectoryName: string): void {
+  setExportDirectoryName(exportDirectoryName?: string): void {
     this.client.run('delete from export_directory_name');
-    this.client.run(
-      `
+    if (exportDirectoryName !== undefined) {
+      this.client.run(
+        `
       insert into export_directory_name (
         export_directory_name
       ) values (?)
       `,
-      exportDirectoryName
-    );
+        exportDirectoryName
+      );
+    }
   }
```

Comment:
Yes thanks for checking!

-----

> Is deleting the directory name on unconfigure already happening?

Yup this is happening. Full unconfigure is great because it just wipes the entire workspace / store - we don't have to explicitly update it every time we add state to the store:

https://github.com/votingworks/vxsuite/blob/a539e3089979d579008f473eb7dfa9c5ef95be08/apps/scan/backend/src/app.ts#L214-L216

https://github.com/votingworks/vxsuite/blob/a539e3089979d579008f473eb7dfa9c5ef95be08/apps/scan/backend/src/util/workspace.ts#L76

https://github.com/votingworks/vxsuite/blob/a539e3089979d579008f473eb7dfa9c5ef95be08/apps/scan/backend/src/store.ts#L196

https://github.com/votingworks/vxsuite/blob/a539e3089979d579008f473eb7dfa9c5ef95be08/libs/db/src/client.ts#L310

https://github.com/votingworks/vxsuite/blob/a539e3089979d579008f473eb7dfa9c5ef95be08/libs/db/src/client.ts#L258-L259

-----

> Does changing the precinct during polls open reset the export directory name? Does it need to?

It doesn't need to because the export directory name is only set _after_ the first ballot is cast and the first CVR is exported, and the precinct can only be changed _before_ polls open and ballot casting. That said, we are still resetting it by virtue of `setPrecinctSelection` calling `workspace.resetElectionSession`, which is where I added the clearing logic:

https://github.com/votingworks/vxsuite/blob/a539e3089979d579008f473eb7dfa9c5ef95be08/apps/scan/backend/src/app.ts#L218-L227